Essentials/src/main/java/com/earth2me/essentials/commands/Commandeditsign.java

```diff
@@ -79,8 +79,11 @@ protected void run(final Server server, final User user, final String commandLab
                     user.sendMessage(tl("editsignCommandClearLine", line + 1));
                 }
             } else if (args[0].equalsIgnoreCase("copy")) {
-                final int line = args.length == 1 ? -1 : Integer.parseInt(args[1]) - 1;
+                if (callSignEvent(sign, user.getBase(), sign.getLines())) {
+                    return;
+                }
```

Comment:
why?

Reply:
I don't think it's a good idea to let players copy signs from places they don't have access to.

Reply:
I think he's referring to that specific code block and not the why of the PR.

Reply:
Copying is not modifying a sign. If we really *really* need to call an event for copying from a sign, it should be a new event.

Reply:
It makes sense, what would be the best approach?

Reply:
no it doesn't, and there's no reason you shouldn't be able to copy a waxed sign. regardless this would fire a `SignChangeEvent` when no modification to a sign it happening.

Reply:
so that protection plugins can cancel the event if the player doesn't have permission in the protection

Reply:
If it's something the Essentials team doesn't think is appropriate, I can remove it.

Reply:
You're allowing protection plugins to block someone from... reading a sign. They can already type out anything that they could copy and paste with `/editsign` - it's nothing more than a convenience tool.

If we do want to call an event here, it *should not* be `SignChangeEvent`. `SignChangeEvent` is also used by block logging plugins and this will spam up logs with "sign changes" where the player didn't actually edit the sign at all.

I'm not convinced letting other plugins block copying is a worthwhile feature at all. If we do support this, it needs a custom event, not `SignChangeEvent`.